benchmark-sets/structured/avahi.yaml

```diff
@@ -0,0 +1,51 @@
+"functions":
+- "name": "avahi_dns_packet_new_reply"
+  "param_names":
+  - "p"
+  - "mtu"
+  - "copy_queries"
+  - "aa"
+  "param_types":
+  - "struct.AvahiDnsPacket *"
```

Comment:
would it make sense to break out tags from types? something like this?

```
"param_types":
- "AvahiDnsPacket *"
- "int"
- "int"
- "int"
"param_tags":
- "struct"
- ""
- ""
- ""

Reply:
Sure! I can implement a temp solution for now and request FI to separate this later.
Ideally we want to keep `introspector.py` as lightweight and transparent as possible.

Reply:
@DavidKorczynski could you please look into this later?
I will have a temporary solution for now so no rush.
But in the long-term, I suppose we do want to avoid having tags in types because they normally do not appear in function signatures?

Reply:
The temp fix is done.
@trashvisor: I discarded the tag (e.g., `struct`) because I presume you don't require it. Please let me know if that can be useful. Thanks!

Reply:
that should be good for now. thanks!

Reply:
Isn't `struct` part of the type?

```c
struct AvahiDnsPacket {
  int a1;
} defedAvahiDnsPacket;
```

Declares two types:

```
struct AvahiDnsPacket
defedAvahiDnsPacket
```

You'll always need to provide `struct` when creating a variable of `struct AvahiDnsPacket`?


I guess my question is what do we get from separating tag? What other tags are there, is it e.g. `const` you have in mind? If so, then there is a difference in that you can have `const struct AvahiDnsPacket` and `struct AvahiDnsPacket`, but, you can't use it in a similar manner with structs, e.g. `AvahiDnsPacket` and `struct AvahiDnsPacket` -- only the later is correct use.

> I discarded the tag (e.g., struct) because I presume you don't require it.

Not sure I understand the context, but I wouldn't have thought `AvahiDnsPacket` be used without `struct`?

Reply:
I think @trashvisor was looking for a symbol name that is consistent with the one used in the signature.
In this case, the signature is [`AvahiDnsPacket* avahi_dns_packet_new_reply(AvahiDnsPacket* p, unsigned mtu, int copy_queries, int aa);`](https://github.com/avahi/avahi/blob/99eae8dd997233d3e7cff7254275d5abcaae53a9/avahi-core/dns.h#L44), which does not have the `struct` tag.

Hence from a user perspective, by looking at the function signature, the user cannot map `AvahiDnsPacket` to `struct.AvahiDnsPacket`.

Reply:
> You'll always need to provide struct when creating a variable of struct AvahiDnsPacket?

Please correct me if I am wrong, but I thought we can omit the `struct`, for example:
https://github.com/avahi/avahi/blob/99eae8dd997233d3e7cff7254275d5abcaae53a9/avahi-core/dns.c#L858

Because the author used the same name:
https://github.com/avahi/avahi/blob/99eae8dd997233d3e7cff7254275d5abcaae53a9/avahi-core/dns.h#L32C1-L36C18

Reply:
> So, in this case I think it's correct to omit struct from the type/tag, since it's a typedef

Could FI omit the tag in this and similar cases? It allows better consistency with the function from developers' view (e.g., the return/param types are `AvahiDnsPacket`, not `struct AvahiDnsPacket`) and can avoid confusing users/LLM.

>  in the event that there are no typedef then I think struct should be part of the type and not a tag

Would it make sense for FI to output `struct AvahiDnsPacket` (i.e., replace dot with a space) in those cases, because that's the exact words that users/LLMs will use in fuzz targets?